File: flipperf/src/main/java/com/flipkart/flipperf/newlib/toolbox/PreferenceManager.java
package com.flipkart.flipperf.newlib.toolbox;

import android.content.Context;
import android.content.SharedPreferences;

/**
 * Preference Manager to store network speed
 */
public class PreferenceManager {

    private static final String PREFERENCES = "PREFERENCES";
    private final SharedPreferences mSharedPreferences;

    public PreferenceManager(Context context) {
        this.mSharedPreferences = context.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE);
    }

    public void setAverageSpeed(String networkType, float avgSpeed) {
        SharedPreferences.Editor editor = mSharedPreferences.edit();
        editor.putFloat(networkType, avgSpeed);
        editor.apply();
    }

    public float getAverageSpeed(String networkType) {
        return this.mSharedPreferences.getFloat(networkType, 0F);
    }
}

File: library-kotlin/src/main/java/com/flipkart/okhttpstats/kotlin/toolbox/PreferenceManager.kt
package com.flipkart.okhttpstats.kotlin.toolbox

import android.content.Context
import android.content.SharedPreferences

/**
 * Preference Manager to store network speed
 */
class PreferenceManager(context: Context) {
    private val sharedPreferences: SharedPreferences

    init {
        this.sharedPreferences = context.getSharedPreferences(PREFERENCES, Context.MODE_PRIVATE)
    }

    fun setAverageSpeed(networkType: String, avgSpeed: Float) {
        val editor = sharedPreferences.edit()
        editor.putFloat(networkType, avgSpeed)
        editor.apply()
    }

    fun getAverageSpeed(networkType: String): Float {
        return this.sharedPreferences.getFloat(networkType, 0f)
    }

    companion object {
        private val PREFERENCES = "PREFERENCES"
    }
}